package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,340 @@
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.variance = self.variance[:self.n_components]
+        self.p_components = e_vects[:self.n_components, sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+        self._calculated = True
+
+    def transform(self, atomgroup, n_components=None, start=None, stop=None,
+                  step=None):
+        """Apply the dimensionality reduction on a trajectory
+
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup/ Universe
+            The atomgroup or universe containing atoms to be PCA transformed.
+        n_components: int, optional
+            The number of components to be projected onto, Default none: maps
+            onto all components.
+        start: int, optional
+            The frame to start on for the PCA transform. Default: None becomes
+            0, the first frame index.
+        stop: int, optional
+            Frame index to stop PCA transform. Default: None becomes n_frames.
+            Iteration stops *before* this frame number, which means that the
+            trajectory would be read until the end.
+        step: int, optional
+            Number of frames to skip over for PCA transform. Default: None
+            becomes 1.
+        Returns
+        -------
+        pca_space : array, shape (number of frames, number of components)
+        """
+        if not self._calculated:
+            self.run()
+
+        if isinstance(atomgroup, Universe):
+            atomgroup = atomgroup.atoms
+
+        if(self._n_atoms != atomgroup.n_atoms):
+            raise ValueError('PCA has been fit for'
+                             '{} atoms. Your atomgroup'
+                             'has {} atoms'.format(self._n_atoms,
+                                                   atomgroup.n_atoms))
+        if not (self._atoms.types == atomgroup.types).all():
+            warnings.warn('Atom types do not match with types used to fit PCA')
+
+        traj = atomgroup.universe.trajectory
+        start, stop, step = traj.check_slice_indices(start, stop, step)
+        n_frames = len(range(start, stop, step))
+
+        dim = (n_components if n_components is not None else
+               self.p_components.shape[1])
+
+        dot = np.zeros((n_frames, dim))
+
+        for i, ts in enumerate(traj[start:stop:step]):
+            xyz = atomgroup.positions.ravel() - self.mean
+            dot[i] = np.dot(xyz, self.p_components[:, :n_components])
+
+        return dot
+
+
+def cosine_content(pca_space, i):
+    """Measure the cosine content of the PCA projection.
+
+    Cosine content is used as a measure of convergence for a protein
+    simulation. If this function is used in a publication, please cite
```

Comment:
```
The cosine content of pca projections can be used as an indicator if a simulations is converged. Values close to 1 are an indicator that the simulation isn't converged. For values below 0.7 no statement can be made. If you use this function please cite [BerkHess1]_.
```

The cosine content is not a measure. This also stands in the paper. It can only be used as an indicator.